ui/src/app/views/cohort-list/component.tsx

```diff
@@ -0,0 +1,147 @@
+import {Component} from '@angular/core';
+
+import {ResourceCard} from 'app/views/resource-card/component';
+
+import {
+  RecentResource,
+  WorkspaceAccessLevel
+} from 'generated/fetch';
+
+import {convertToResources, ResourceType} from 'app/utils/resourceActionsReact';
+
+import {cohortsApi} from 'app/services/swagger-fetch-clients';
+
+import {ClrIcon} from 'app/components/icons';
+
+import {TooltipTrigger} from 'app/components/popups';
+import {SpinnerOverlay} from 'app/components/spinners';
+
+import {CardButton} from 'app/components/buttons';
+import {navigate} from 'app/utils/navigation';
+import * as React from 'react';
+
+import {FadeBox} from 'app/components/containers';
+import {WorkspaceData} from 'app/resolvers/workspace';
+import {reactStyles, ReactWrapperBase, withCurrentWorkspace} from 'app/utils/index';
+
+const styles = reactStyles({
+  pageArea: {
+    display: 'flex',
+    justifyContent: 'flex-start'
+  },
+
+  cohortsHeader: {
+    color: '#2F2E7E',
+    fontSize: '20px',
+    lineHeight: '24px',
+    fontWeight: 600,
+    marginTop: 0,
+    paddingTop: '0.5rem'
+  },
+
+  resourceCardArea: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  },
+
+  infoIcon: {
+    color: '#2691D0',
+    cursor: 'pointer',
+    marginLeft: '0.2rem',
+    height: '16px',
+    width: '16px'
+  },
+  addCard: {
+    width: 200, height: 105,
+    marginTop: '1rem', marginRight: '1rem',
+    fontSize: 18, fontWeight: 500, lineHeight: '22px',
+  }
+});
+
+export const CohortList = withCurrentWorkspace()(
+  class extends React.Component<{workspace: WorkspaceData},
+      {cohortList: RecentResource[], cohortsLoading: boolean}> {
+    constructor(props) {
+      super(props);
+      this.state = {
+        cohortsLoading: false,
+        cohortList: undefined
+      };
+    }
+
+    componentDidMount() {
+      this.reloadCohorts();
+    }
+
+    async reloadCohorts() {
+      try {
+        const {namespace, id, accessLevel} = this.props.workspace;
+        this.setState({
+          cohortsLoading: true
+        });
+        const resp = await cohortsApi().getCohortsInWorkspace(namespace, id);
+        this.setState({
+          cohortList: convertToResources(resp.items, namespace,
+            id, accessLevel as unknown as WorkspaceAccessLevel, ResourceType.COHORT),
```

Comment:
nice! does this get around the casting for the fetch resource types?

Reply:
Sort of. It handles it at a single level, but nested values I think still cause a problem. But that's the syntax around it.